packages/router/src/router-context.tsx

```diff
@@ -30,6 +30,8 @@ export interface RouterSetContextProps {
   setState: (newState: Partial<RouterState>) => void
 }
 
+// TODO what is this SetContext for???
```

Comment:
@Tobbe any idea what this context here is for? Like why is it split up into two contexts?

Just needs a comment to clarify here I think

Reply:
<img width="376" alt="image" src="https://github.com/redwoodjs/redwood/assets/30793/3505d802-e385-4956-be13-806ecab681a7">

This is the key.
There's a separate "setter" context and "getter" context. I did that to optimize the number of redraws. (Might even have been an infinite loop of updates. Can't fully remember)

The idea comes from this article https://kentcdodds.com/blog/how-to-optimize-your-context-value